### PR TITLE
CORE-5033 refactor of start flow endpoint definition

### PR DESCRIPTION
**Pull request by LWogan**
change create flow endpoint to a POST from PUT as it is not idempotent.
move clientrequest id param to request body, 
throw error when trying to create the same flow twice (same client request id). return status code 409 (conflict)

### CODE REVIEW

File: processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json [hunk 1]
```diff
@@ -2279,43 +2249,43 @@
...
-            "nullable" : true,
+            "nullable" : false,
             "example" : "string"
           }
         },
         "nullable" : false
       },
       "CpiIdentifier_1" : {
-        "required" : [ "name", "signerSummaryHash", "version" ],
+        "required" : [ "cpiName", "cpiVersion" ],
```
Comment (vkolomeyko) [inline processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json]: This change is somewhat unexpected...
This is probably down to the fact that we have **two** `CpiIdentifier` data classes in OpenAPI.

@josephzunigadaly @mbrkic-r3 ^

File: components/flow/flow-rpcops-service/src/main/kotlin/net/corda/flow/rpcops/v1/types/request/HTTPStartFlowRequest.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * Request sent to start a flow
+ *
+ * @param clientRequestId Client provided flow identifier
+ * @param flowClassName Fully qualified class name of the flow to start.
+ * @param flowParams Optional start arguments string passed to the flow. Defaults to empty string.
+ */
+data class HTTPStartFlowRequest(
+    val clientRequestId: String,
+    val flowClassName: String,
+    val flowParams: String
```
Comment (lankydan) [inline components/flow/flow-rpcops-service/src/main/kotlin/net/corda/flow/rpcops/v1/types/request/HTTPStartFlowRequest.kt]: Should we name this `requestData` to match the `RPCRequestData` object that this string is represented by in `RPCStartableFlow`?

File: components/flow/flow-rpcops-service/src/main/kotlin/net/corda/flow/rpcops/v1/types/response/HTTPFlowStatusResponse.kt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package net.corda.flow.rpcops.v1.response
+package net.corda.flow.rpcops.v1.types.response
```
Comment (lankydan) [inline components/flow/flow-rpcops-service/src/main/kotlin/net/corda/flow/rpcops/v1/types/response/HTTPFlowStatusResponse.kt:1]: Is there much benefit to moving the packages? Hard to see in github.

File: processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json [hunk 4]
```diff
@@ -459,18 +459,18 @@
...
-          "required" : true,
-          "schema" : {
-            "type" : "string",
-            "description" : "Client provided flow identifier",
-            "nullable" : false,
-            "example" : "string"
-          }
         } ],
+        "requestBody" : {
+          "description" : "requestBody",
```
Comment (lankydan) [inline processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json:499]: Is this right? We don;t have a `requestBody` field anymore?

Comment (sprintengine) [reply processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json:499]: Looks like there was a reordering of the two endpoints but I think its correct i.e. /flow/{holdershortid} comes before /flow/{holdershortid}/{clientrequestid}

File: processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json (hunk 1 above)
Comment (LWogan) [reply processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json]: Ive reverted these changes. The tests fail with them present. However they are what is generated in https://localhost:8888/api/v1/swagger.json

File: components/flow/flow-rpcops-service/src/main/kotlin/net/corda/flow/rpcops/v1/types/response/HTTPFlowStatusResponse.kt (hunk 3 above)
Comment (LWogan) [reply components/flow/flow-rpcops-service/src/main/kotlin/net/corda/flow/rpcops/v1/types/response/HTTPFlowStatusResponse.kt:1]: brought it in line with other projects. although my primary motivation was to reduce differences between other working endpoints when mine was broken due to https://r3-cev.atlassian.net/browse/CORE-5397

File: processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json (hunk 4 above)
Comment (LWogan) [reply processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json:499]: requestBody isn't a field, its a type of object here. The endpoint has a request body which is the HTTPStartFlowRequest class although this generated the older name. Bit strange, I am certain i published the RPC worker before generating the swagger. Ill do a clean and publish all images and see if that resolves it. There is some other strange output of diffs that arent mine too.

File: components/flow/flow-rpcops-service/src/main/kotlin/net/corda/flow/rpcops/v1/types/request/HTTPStartFlowRequest.kt (hunk 2 above)
Comment (LWogan) [reply components/flow/flow-rpcops-service/src/main/kotlin/net/corda/flow/rpcops/v1/types/request/HTTPStartFlowRequest.kt]: makes sense, ill update

File: processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json (hunk 4 above)
Comment (LWogan) [reply processors/rpc-processor/src/integrationTest/resources/swaggerBaseline.json:499]: Interesting. I did a full clean rebuild and it still outputted this in the generated swagger. There are no types or fields named StartFlowRequest in the code

```"requestBody" : {
          "description" : "requestBody",
          "content" : {
            "application/json" : {
              "schema" : {
                "$ref" : "#/components/schemas/StartFlowRequest"
              }
            }
          },
          "required" : true
        },
```

```
"StartFlowRequest" : {
        "required" : [ "httpStartFlow" ],
        "properties" : {
          "httpStartFlow" : {
            "$ref" : "#/components/schemas/HTTPStartFlowRequest"
          }
        },
        "description" : "StartFlowRequest",
        "nullable" : false
```
.
```
      "HTTPStartFlowRequest" : {
        "required" : [ "clientRequestId", "flowClassName", "flowParams" ],
        "type" : "object",
        "properties" : {
          "clientRequestId" : {
            "type" : "string",
            "nullable" : false,
            "example" : "string"
          },
          "flowClassName" : {
            "type" : "string",
            "nullable" : false,
            "example" : "string"
          },
          "flowParams" : {
            "type" : "string",
            "nullable" : false,
            "example" : "string"
          }
        },
        "description" : "Information required to start a flow for this holdingId",
        "nullable" : false
      }
```

**Review by JamesHR3 — COMMENTED** [review]
We're going to need to do a bit of comms before we merge this, as it will significantly impact the workflow of anyone executing flows as part of their day to day dev work.

**Review by JamesHR3 — APPROVED** [review]

**Pull request merged**